scheduler/pkg/kafka/gateway/infer.go

```diff
@@ -372,13 +372,18 @@ func (kc *InferKafkaHandler) AddModel(modelName string) error {
 	inputTopic := kc.topicNamer.GetModelTopicInputs(modelName)
 	outputTopic := kc.topicNamer.GetModelTopicOutputs(modelName)
 	if err := kc.createTopics([]string{inputTopic, outputTopic}); err != nil {
+		// we must delete the model as we've failed to create the topics, scheduler will re-issue a create cmd, if model
+		// exists in loadedModels model-gw will respond with success and it'll be no-op and topics won't be retried to
+		// be created and scheduler will presume model loaded successfully
+		delete(kc.loadedModels, modelName)
 		return err
 	}
 
 	kc.subscribedTopics[inputTopic] = true
 	err := kc.subscribeTopics()
 	if err != nil {
 		kc.logger.WithError(err).Errorf("failed to subscribe to topics")
+		// why do we not return an error here? does confluent pkg keep retrying in background?
```

Comment:
The kafka library does indeed keep retrying in the background for situations such as network issues, subscribed topics that don't yet exist, etc. However, that is also why subscribeTopics (well, the `kc.consumer.subscribeTopics` called within subscribeTopics) only returns an error in the case of a fatal consumer error (or badly specified topic names). For reference, the return error cases are [the ones specified](https://docs.confluent.io/platform/current/clients/librdkafka/html/rdkafka_8h.html#a0ebe15e9d0f39ccc84e9686f0fcf46f1) for `rd_kafka_subscribe` in librdkafka. So I think we _should_ return an error, but we were getting away with it so far because this path was never really exercised in practice.